dockerfiles/hpcc/templates/thor.yaml

```diff
@@ -0,0 +1,128 @@
+{{ range $thor := $.Values.thor -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .name | quote }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: {{ .name | quote }}
+  template:
+    metadata:
+      labels:
+        run: {{ .name | quote }}
+    spec:
+      containers:
+      - name: {{ .name | quote }}
+        ports:
+          - containerPort: {{ $thor.slaveport }}
+        args: [
+                {{ include "hpcc.utils.configArg" . }},
+                {{ include "hpcc.utils.daliArg" $ }},
+                --masterport={{ $thor.masterport }},
+                --numSlaves={{ $thor.numSlaves }}
+              ]
+{{ include "hpcc.utils.addImageAttrs" (dict "root" $ "imagename" "thormaster") | indent 8 }}
+{{ include "hpcc.utils.addVolumeMounts" . | indent 8 }}
+{{ include "hpcc.utils.addVolumes" . | indent 6 }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ printf "%s-slave" .name | quote }}
+spec:
+  replicas: {{ $thor.numSlaves }}
+  selector:
+    matchLabels:
+      run: {{ printf "%s-slave" .name | quote }}
+  template:
+    metadata:
+      labels:
+        run: {{ printf "%s-slave" .name | quote }}
+    spec:
+      {{ if $.Values.global.singleNode | default false -}}
+      initContainers:
+      # This is a bit of a hack, to ensure that the persistent storage mounted for dllserver
+      # is writable. This is not something we would want to do if using anything other than
+      # local storage (which is only sensible on single-node systems).
+      - name: volume-mount-hack
+        image: busybox
+        command: ["sh", "-c", "chown -R 999:1000 /var/lib/HPCCSystems/hpcc-data"]
```

Comment:
Should do chmod 755 rather than chown, I suspect. That way don't need to rely on magic user/gruop ids that may or may not be right in the future.

Should also consider templatizing as we are going to need this for any place we want persistent storage.

Reply:
I've templatized it, and will check chmod 755 works as expected